src/components/connectAdvanced.js

```diff
@@ -100,203 +90,169 @@ export default function connectAdvanced(
-        // this is especially important for 'ref' since that's a reference back to the component
-        // instance. a singleton memoized selector would then be holding a reference to the
-        // instance, preventing the instance from being garbage collected, and that would be bad
-        const withExtras = { ...props }
-        if (withRef) withExtras.ref = this.setWrappedInstance
-        if (renderCountProp) withExtras[renderCountProp] = this.renderCount++
-        if (this.propsMode && this.subscription) withExtras[subscriptionKey] = this.subscription
-        return withExtras
+        return <WrappedComponent {...derivedProps} />
       }
 
       render() {
-        const selector = this.selector
-        selector.shouldComponentUpdate = false
-
-        if (selector.error) {
-          throw selector.error
-        } else {
-          return createElement(WrappedComponent, this.addExtraProps(selector.props))
+        if (this.props.unstable_observedBits) {
+          return (
+            <Consumer unstable_observedBits={this.props.unstable_observedBits}>
+              {this.renderWrappedComponent}
+            </Consumer>
+          )
         }
+        return (
+          <Consumer>
+            {this.renderWrappedComponent}
+          </Consumer>
+        )
       }
     }
 
-    /* eslint-enable react/no-deprecated */
-
     Connect.WrappedComponent = WrappedComponent
     Connect.displayName = displayName
-    Connect.childContextTypes = childContextTypes
-    Connect.contextTypes = contextTypes
-    Connect.propTypes = contextTypes
-
-    if (process.env.NODE_ENV !== 'production') {
-      Connect.prototype.componentWillUpdate = function componentWillUpdate() {
-        // We are hot reloading!
-        if (this.version !== version) {
-          this.version = version
-          this.initSelector()
-
-          // If any connected descendants don't hot reload (and resubscribe in the process), their
-          // listeners will be lost when we unsubscribe. Unfortunately, by copying over all
-          // listeners, this does mean that the old versions of connected descendants will still be
-          // notified of state changes; however, their onStateChange function is a no-op so this
-          // isn't a huge deal.
-          let oldListeners = [];
-
-          if (this.subscription) {
-            oldListeners = this.subscription.listeners.get()
-            this.subscription.tryUnsubscribe()
-          }
-          this.initSubscription()
-          if (shouldHandleStateChanges) {
-            this.subscription.trySubscribe()
-            oldListeners.forEach(listener => this.subscription.listeners.subscribe(listener))
-          }
-        }
+    if (forwardRef) {
+      Connect.prototype.renderWrappedComponent = Connect.prototype.renderWrappedComponentWithRef
+      Connect.propTypes = {
+        props: propTypes.object,
+        forwardRef: propTypes.oneOfType([
```

Comment:
If we're doing `forwardRef`, then this isn't even the actual exposed component - I don't think proptypes are necessary in that case.